classes/controlledVocab/ControlledVocab.php

```diff
@@ -1,108 +1,146 @@
 <?php
-/**
- * @defgroup controlled_vocab Controlled Vocabulary
- */
 
 /**
  * @file classes/controlledVocab/ControlledVocab.php
  *
- * Copyright (c) 2014-2021 Simon Fraser University
- * Copyright (c) 2000-2021 John Willinsky
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
  * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
  *
  * @class ControlledVocab
  *
- * @ingroup controlled_vocab
- *
- * @see ControlledVocabDAO
- *
- * @brief Basic class describing an controlled vocab.
+ * @brief ControlledVocab model class
  */
 
 namespace PKP\controlledVocab;
 
-use PKP\db\DAORegistry;
+use Eloquence\Behaviours\HasCamelCasing;
+use Illuminate\Database\Query\JoinClause;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Database\Eloquent\Builder;
+use PKP\facades\Locale;
 
-class ControlledVocab extends \PKP\core\DataObject
+class ControlledVocab extends Model
 {
-    //
-    // Get/set methods
-    //
+    use HasCamelCasing;
+
+    protected $table = 'controlled_vocabs';
+    protected $primaryKey = 'controlled_vocab_id';
+
+    protected $guarded = [
+        'controlled_vocab_id',
+    ];
 
     /**
-     * get assoc id
+     * Indicates if the model should be timestamped.
      *
-     * @return int
+     * @var bool
      */
-    public function getAssocId()
+    public $timestamps = false;
+
+    protected function casts(): array
     {
-        return $this->getData('assocId');
+        return [
+            'symbolic' => 'string',
+            'assoc_type' => 'integer',
+            'assoc_id' => 'integer',
+        ];
     }
 
     /**
-     * set assoc id
-     *
-     * @param int $assocId
+     * Accessor and Mutator for primary key => id
      */
-    public function setAssocId($assocId)
+    protected function id(): Attribute
     {
-        $this->setData('assocId', $assocId);
+        return Attribute::make(
+            get: fn($value, $attributes) => $attributes[$this->primaryKey] ?? null,
+            set: fn($value) => [$this->primaryKey => $value],
+        );
     }
 
     /**
-     * Get associated type.
+     * Compatibility function for including note IDs in grids.
      *
-     * @return int
+     * @deprecated 3.5.0 Use $model->id instead. Can be removed once the DataObject pattern is removed.
      */
-    public function getAssocType()
+    public function getId(): int
     {
-        return $this->getData('assocType');
+        return $this->id;
     }
 
     /**
-     * Set associated type.
+     * Compatibility function for including notes in grids.
      *
-     * @param int $assocType
+     * @deprecated 3.5. Use $model or $model->$field instead. Can be removed once the DataObject pattern is removed.
      */
-    public function setAssocType($assocType)
+    public function getData(?string $field): mixed
```

Comment:
Is `getData(null)` ever called in practice? If not, I'd rather forestall that calling convention.